### PR TITLE
use cmake target-interface for easier use of lib

### PR DESCRIPTION
**Pull request by ObiWahn**
https://www.youtube.com/watch?v=bsXLMQ6WgIk - talk on modern cmake

### CODE REVIEW

**Comment by odinthenerd** [comment]
Looks good but I'm not the cmake expert, @chieltbest what do you think?

**Comment by chieltbest** [comment]
This seems good, we've been using this sort of style for some time now in mpl, so I don't see any problem with it.

**Comment by LegalizeAdulthood** [comment]
Why leave the old stuff present in the file but commented out?

File: CMakeLists.txt [hunk 1]
```diff
@@ -12,29 +12,31 @@ if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
...
+#    Lib/Register/Types.hpp
+#    Lib/Register/Utility.hpp
+#)
+#
+#add_library(Kvasir ${MPL_GROUP} ${REGISTER_GROUP})
+
+add_library(Kvasir INTERFACE)
+target_include_directories(Kvasir INTERFACE ${CMAKE_CURRENT_SOURCE_DIR}/Lib)
+#set_target_properties(Kvasir PROPERTIES LINKER_LANGUAGE CXX)
+target_compile_definitions(Kvasir INTERFACE "DBRIGAND_NO_BOOST_SUPPORT")
```
Comment (LegalizeAdulthood) [inline CMakeLists.txt:39]: This doesn't seem right, shouldn't the line be

    target_compile_definitions(Kvasir INTERFACE BRIGAND_NO_BOOST_SUPPORT)

Or does the gating symbol really begin with "D"?

**Comment by LegalizeAdulthood** [comment]
Having removed all the source files from the library target, it means that generating a project for an IDE won't show any files in the project.  I think this is a mistake.

**Pull request closed without merging**